packages/nimble/R/genCpp_sizeProcessing.R

```diff
@@ -991,6 +1000,23 @@ sizeNimbleListReturningFunction <- function(code, symTab, typeEnv) {
   if(length(asserts) == 0) NULL else asserts
 }
 
+sizeOptim <- function(code, symTab, typeEnv) {
+    # TODO input an optional nimbleList for control
+    asserts <- nimble:::recurseSetSizes(code, symTab, typeEnv)
+    code$type <- 'nimbleList'
+    code$sizeExprs <- symTab$getSymbolObject('OptimResultNimbleList', inherits = TRUE)
+    code$toEigenize <- 'maybe'
+    code$nDim <- 0
+    asserts <- c(asserts, sizeInsertIntermediate(code$caller, code$callerArgID, symTab, typeEnv))
```

Comment:
Oh perfect! Thanks for pointing me to your branch. I'll grab your changes and incorporate them into this PR.